src/python/pants/option/subsystem.py

```diff
@@ -119,9 +123,65 @@ def __eq__(self, other: Any) -> bool:
         return bool(self.options == other.options)
 
 
-_T = TypeVar("_T", bound=Subsystem)
+_SubsystemT = TypeVar("_SubsystemT", bound=Subsystem)
+_T = TypeVar("_T")
 
 
-async def _construct_subsytem(subsystem_typ: type[_T]) -> _T:
+class Option(Generic[_T]):
+    """Data-descriptor for subsystem options.
+
+    This class exists to help eliminate the repitition of declaring options in `register_options`
+    and having to declare properties for mypy's sake.
+
+    Usage:
+        class Engine(Subsystem):
+            ...
+
+            cylinders = Option[int]("--cylinders", default=6, help="...")
+
+        ...
+
+        engine: Engine = ...
+        engine.cylinders  # mypy knows this is an int
+
+    Under-the-hood:
+        - The `type` argument if omitted defaults to the type of `default`
```

Comment:
I think this is probably a good idea, but how would you feel about waiting until a followup PR to implement this so that we can properly consider it? 

I have some concerns that it's confusing how the type hint `Option[int]` already is expressing the type, so I naively thought you were doing fancy parsing of the MyPy types. The `default` thing wasn't intuitive to me.

I still think the `default` thing is probablyyyy a good idea, but it's the most controversial part of this PR fwict.

Reply:
Works for me!